helper/convert/type.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package convert
+
+import (
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+// An private version of this function to convert cty.Type to tftypes.Type exists in
+// internal/plugin/convert/schema.go but this is a simplified and public version
+// that maps the types one to one without any error handling, under the assumption
+// the incoming data will always be valid
+func ToTfType(in cty.Type) tftypes.Type {
+	switch {
+	case in.IsPrimitiveType():
```

Comment:
For completeness, the final primitive that we'd need to add here is `DynamicPseudoType`, which should never actually show up for the use-cases we have, but it's a pretty simple add

```go
	case in.IsPrimitiveType():
		if in == cty.DynamicPseudoType {
			return tftypes.DynamicPseudoType
		}
```